package.json

```diff
@@ -11,14 +11,15 @@
     "build:es": "rm -rf es && cross-env BABEL_ENV=es babel src -d es --extensions .ts,.tsx --ignore src/**/*.test.tsx",
     "build:ts": "tsc --rootDir src -d --declarationDir lib && tsc --rootDir src -d --declarationDir es",
     "build:icon": "rm -rf src/General/Icon/components && babel-node --extensions .ts,.tsx src/Utils/Icon/generateIcons.ts && npm run format:icon",
+    "check:branch": "if [ \"$(git rev-parse --abbrev-ref HEAD)\" = v4 ] ; then echo Branch check passed. ; else echo Error: Not on branch v4. ; exit 1 ; fi",
     "format:icon": "prettier --write src/General/Icon/components/*.tsx",
     "lint": "npm run lint:css && npm run lint:ts",
     "lint:css": "stylelint src/Style/**/*.ts",
     "lint:ts": "eslint src/**/*.{ts,tsx}",
-    "preversion": "npm whoami && npm run validate",
+    "preversion": "npm run check:branch && git pull && npm whoami && npm run validate",
     "postversion": "npm publish",
-    "prepublishOnly": "npm whoami && npm run validate",
-    "postpublish": "git push origin --tags",
+    "prepublishOnly": "npm run check:branch && git pull && npm whoami && npm run validate",
```

Comment:
why preversion and  prepublishOnly do similar things?

Reply:
Prevent someone who doesn't use `npm version` script to do publish from skipping the validation process.

Reply:
how do you prevent `npm publish` without an updated version in package.json?

Reply:
ah, then publish will not work because version already exist.

Reply:
`npm` will deny the publish without version change.

Reply:
please add instruction to wiki